### PR TITLE
[sdk] Adds experimental section to sdk

### PR DESCRIPTION
**Pull request by damirka**
This PR isolates BCS and libraries for it in the `experimental` section. If it is required, we could even remove it from SDK's default imports.

### CODE REVIEW

**Review by gegaowp — APPROVED** [review]
Looks great, this will be very useful!
wondering about `experimental`, is there some specific level of stability we need to reach before making it `prod` ?

File: sdk/typescript/src/experimental/types.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+initializers.push((b: typeof BCS) =>
+  b.registerStructType("PublishTx", {
+    modules: "vector<vector<u8>>",
+  })
+);
+
+// ========== Move Call Tx ===========
+
+/**
+ * An argument for the transaction. It is a 'meant' enum which expects to have
```
Comment (stella3d) [inline sdk/typescript/src/experimental/types.ts:103]: what does this mean ? unsure whether users will commonly be able to interpret this

File: sdk/typescript/src/experimental/types.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+);
+
+export type StructTag = {
+  address: string,
+  module: string,
+  name: string,
+  typeParams: TypeTag[]
+};
+
+export type TypeTag =
```
Comment (stella3d) [inline sdk/typescript/src/experimental/types.ts:141]: should we have this in the non-experimental section ? this seems to mirror what we have in the rust side already.

File: sdk/typescript/src/experimental/types.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    })
+);
+
+/**
+ * The TransactionData to be signed and sent to the Gateway service.
+ *
+ * Field `sender` is made optional as it can be added during the signing
+ * process and there's no need to define it sooner.
+ */
+export type TransactionData = {
```
Comment (stella3d) [inline sdk/typescript/src/experimental/types.ts:229]: what's the difference between this and the regular `TransactionData` that's on main already ?
I see some minor formatting changes but the purpose and data seem to largely be the same.

**Review by stella3d — COMMENTED** [review]
My only big question is what makes some of the included code experimental vs not, and whether we're duplicating things found in the main TS sdk.
It seems like some things could be unified, but i may be missing context on why they are that way.

**Comment by damirka** [comment]
Closing this due to publish of the `@mysten/bcs`. Will revisit later.

**Pull request closed without merging**